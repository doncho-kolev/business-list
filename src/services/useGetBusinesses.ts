import { useQuery, gql } from "@apollo/client";
import { RestLink } from 'apollo-link-rest';
import { Business } from "../models/Business";

export const BusinessLink = new RestLink({ uri: "https://feinterviewtask.azurewebsites.net/b/6231abada703bb67492d2b8f" });

const getBusinesses = gql`
  query letBusinesses {
    businesses @rest(type: "Business", path: "/") {
      name,
      description,
      address,
      id,
      phone,
      image,
      email
      }
  }
`;

export function useGetBisunesses() {
    const { loading, error, data } = useQuery(getBusinesses);
    return {
        loading,
        error,
        businesses: data?.businesses as Business[]
    }
}