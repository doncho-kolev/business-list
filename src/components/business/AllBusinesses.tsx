import React from 'react';
import { useGetBisunesses } from '../../services/useGetBusinesses';
import BusinessList from './BusinessList';
import { Row } from '../../features/list/Row';
import '../../assets/styles/allBusinesses.scss';


export default function AllBusiness() {
    const { loading, error, businesses } = useGetBisunesses();
    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error</p>;

    return (
        <div className={"container businesses-container"}>
            <div className={"scroll"}>
                <Row
                    cells={[
                        { label: "Name", className: "name-cell" },
                        { label: "Description", className: "description-cell" }
                    ]}
                    className={"header-row"}
                />
                <BusinessList
                    businesses={businesses}
                    valueGetter={b => b.description ?? ''}
                />
            </div>
        </div>
    );
}
