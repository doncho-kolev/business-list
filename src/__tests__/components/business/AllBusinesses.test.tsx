import React from 'react';
import { render, screen } from '@testing-library/react';
import AllBusinesses from '../../../components/business/AllBusinesses';
import * as GetBusinessesService from '../../../services/useGetBusinesses';

jest.mock('../../../components/business/BusinessList', () => ()=> {
return <div>Business List</div>
})

describe("All Businesses List", () => {

    test('renders all businesses component', () => {
        jest.spyOn(GetBusinessesService, "useGetBisunesses").mockImplementation(() => ({
            businesses: [{
                name: "test",
                description: "description for test"
            }]
        }))
        const mount = render(<AllBusinesses />);
        expect(mount.asFragment().children[0]).toMatchSnapshot();
    });
});
