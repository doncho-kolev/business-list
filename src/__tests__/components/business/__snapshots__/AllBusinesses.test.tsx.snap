// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Businesses List renders all businesses component 1`] = `
<div
  class="container businesses-container"
>
  <div
    class="scroll"
  >
    <div
      class="header-row row"
    >
      <div
        class="name-cell cell"
      >
        Name
      </div>
      <div
        class="description-cell cell"
      >
        Description
      </div>
    </div>
    <div>
      Business List
    </div>
  </div>
</div>
`;
